tsdb/points.go

```diff
@@ -928,7 +928,7 @@ func (p Fields) MarshalBinary() []byte {
 			// ensure there is a decimal in the encoded for
 
 			val := []byte(strconv.FormatFloat(t, 'f', -1, 64))
-			hasDecimal := t-float64(int64(t)) > 0
+			hasDecimal := t-float64(int64(t)) != 0
```

Comment:
Should be:
```
_, frac := math.Modf(t)
hasDecimal := frac != 0
```
Because truncating won't work for all values.  http://play.golang.org/p/rdJCCMf3uB